tracer/test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs

```diff
@@ -426,5 +425,12 @@ public string GetTargetFramework()
 
             return $"net{_major}{_minor}{_patch ?? string.Empty}";
         }
+
+        public MockTracerAgent GetMockAgent(bool useStatsD = false)
+        {
+            var agentPort = TcpPortProvider.GetOpenPort();
+            _output.WriteLine($"Assigning port {agentPort} for the agentPort.");
+            return new MockTracerAgent(agentPort, useStatsd: useStatsD);
```

Comment:
This has bugged me for some time. There's no guarantee that the port we pick initially is the port we assign 
```suggestion
            var agentPort = TcpPortProvider.GetOpenPort();
            var agent = new MockTracerAgent(agentPort, useStatsd: useStatsD);
            _output.WriteLine($"Assigned port {agent.Port} for the agentPort.");
            return agent;
```